### PR TITLE
Adding a join function.

### PR DESCRIPTION
**Pull request by tcoupland**
Isn't particularly hardened, but should suit our needs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 A helper library for writing models in the style of Witan workspaces.
 
-## Defining Functions, Predicates and Models
+## API
```
Comment (acron0) [inline README.md:7]: Y U REK ME?

File: README.md [hunk 2]
```diff
@@ -78,7 +78,41 @@ Use this macro to define a workflow for a model. All model workflows defined thi
    :task4 :out)
 ```
 
-## Helper Functions
```
Comment (acron0) [inline README.md]: Y U REK ME 2?

File: README.md [hunk 3]
```diff
@@ -110,8 +144,8 @@ Use this macro to simulate a loop which executes at least once. First arg is a p
 (do-while-> (number-lt-10)
          {:number 1}
          (inc-loop) ;; returns {:number (inc number)}
-         (double-loop)) ;; returns {:number (* 2 number)}
-=> {:number 10}
+         (double-loop) ;; returns {:number (* 2 number)}
+=> {:number 14}
```
Comment (acron0) [inline README.md]: GAH

File: README.md (hunk 1 above)
Comment (tcoupland) [inline README.md:7]: what the heck is going on?!

File: README.md (hunk 2 above)
Comment (tcoupland) [inline README.md]: what the heck is going on?!

Comment (tcoupland) [inline README.md]: ah

Comment (tcoupland) [inline README.md]: should'd just merged the ruddy thing!

**Comment by acron0** [comment]
:shipit:

**Pull request merged**